lib/graphql/query.rb

```diff
@@ -135,6 +135,11 @@ def initialize(schema, query_string = nil, query: nil, document: nil, context: n
       end
     end
 
+    # If a document was provided to `GraphQL::Schema#execute` instead of the raw query string, we will need to get it from the document
+    def query_string
+      @query_string || (document ? document.to_query_string : nil)
```

Comment:
Could this be `||=`, so that we cache the resulting string? If we _can_ cache it, it'd be nice to do it so that subsequent calls to `query_string` to incur any overhead.

Reply:
It seems like a safe thing to cache to me - updated!